### DISCUSSION TITLE
Gosund UP111 - LEDs

### DISCUSSION THREAD

**Discussion by chrisdaniels** [RavenSystem/esp-homekit-devices#2120] [MEPLHAA Scripts]
Hi, I'm struggling getting the LEDs to display Blue for On, and Red for Off on this plug - can anyone assist?

Pinout is here https://templates.blakadder.com/gosund_UP111.html

Currently the LEDs just remain Blue no matter what the status is.

My working JSON:

{
  "c": { 
    "io": [ [ [ 1, 13, 14 ], 2 ], [ [ 3 ], 6 ] ],
	"l": 13, 
    "b": [ [ 0, 5 ] ]
  },
  "a": [{
    "t": 1,
    "0": { "r": [ [ 14, 0 ] ] },
    "1": { "r": [ [ 14, 1 ] ] },
    "b": [ [ 3 , 0 ] ]
  }]
}

**Comment by Tellicious** [comment]
I do not own this plug, so I cannot test the script, but just try this:

```
{
"c": {
"io": [ [ [ 1, 13, 14 ], 2 ], [ [ 3 ], 6 ] ],
"l": 13,
"b": [ [ 0, 5 ] ]
},
"a": [{
"t": 1,
"0": { "r": [ [ 14],[13,1],[1]]},
"1": { "r": [ [ 14,1],[1,1],[13]]},
"b": [[3,0]]
}]
} 
```

**Reply by chrisdaniels** [reply]
Thank you 🥇 
It doesn't turn red when off, but it does turn the blue led off, which is progress 👍

**Reply by chrisdaniels** [reply]
Actually, its the other way around, it lights Blue when off... but I can see what you have done here, so i'm sure I can work this out 👍

**Reply by Tellicious** [reply]
yep, sorry, I misread Tasmota configuration.

let's remove "l":13 to avoid potential interference with your expected behavior. Then maybe you can try re-adding it at a later point

Try this one:
```json
{
"c": {
"io": [ [ [ 1, 13, 14 ], 2 ], [ [ 3 ], 6 ] ],
"b": [ [ 0, 5 ] ]
},
"a": [{
"t": 1,
"0": { "r": [ [ 14],[1,1],[13]]},
"1": { "r": [ [ 14,1],[13,1],[1]]},
"b": [[3,0]]
}]
} 
```

**Reply by chrisdaniels** [reply]
Superb stuff, thank you very much 👍 
I switched the led behaviour around, but the removal of I:13 has corrected the missing red light 👍 👍 
Your config above works 100% 

Thank you
